samples/startup-script/index.js

```diff
@@ -16,130 +16,123 @@
+    const [vm, operation] = await vmObj.create(config);
+    await operation.promise();
+    const [metadata] = await vm.getMetadata();
 
-  const vm = zone.vm(name);
+    // External IP of the VM.
+    const ip = metadata.networkInterfaces[0].accessConfigs[0].natIP;
+    console.log(`Booting new VM with IP http://${ip}...`);
 
-  vm.create(config)
-    .then(data => {
-      const operation = data[1];
-      return operation.promise();
-    })
-    .then(() => {
-      return vm.getMetadata();
-    })
-    .then(data => {
-      const metadata = data[0];
-
-      // External IP of the VM.
-      const ip = metadata.networkInterfaces[0].accessConfigs[0].natIP;
-      console.log(`Booting new VM with IP http://${ip}...`);
-
-      // Ping the VM to determine when the HTTP server is ready.
-      let waiting = true;
-      const timer = setInterval(
-        ip => {
-          http
-            .get('http://' + ip, res => {
-              const statusCode = res.statusCode;
-              if (statusCode === 200 && waiting) {
-                waiting = false;
-                clearTimeout(timer);
-                // HTTP server is ready.
-                console.log('Ready!');
-                callback(null, ip);
-              }
-            })
-            .on('error', () => {
-              // HTTP server is not ready yet.
-              process.stdout.write('.');
-            });
-        },
-        2000,
-        ip
-      );
-    })
-    .catch(err => callback(err));
+    // Ping the VM to determine when the HTTP server is ready.
+    await pingVM(ip);
+    return ip;
+  } catch (err) {
+    console.error(`Something went wrong while creating ${name} :`, err);
+  }
 }
 
+async function pingVM(ip) {
+  let waiting = true;
+  while (waiting) {
+    await new Promise(r => setTimeout(r, 2000));
+    try {
+      const res = await axios.get(`http://${ip}`);
+      const statusCode = res.status;
+      if (statusCode === 200) {
+        waiting = false;
+        console.log('Ready!');
+        return;
+      } else {
+        process.stdout.write('.');
+      }
+    } catch (err) {
+      process.stdout.write('.');
+    }
+  }
+}
 // List all VMs and their external IPs in a given zone.
-// callback(error, [[name, ip], [name, ip], ...])
-function listVms(callback) {
-  zone
-    .getVMs()
-    .then(data => {
-      const vms = data[0];
-      const results = vms.map(vm => vm.getMetadata());
-      return Promise.all(results);
-    })
-    .then(res =>
-      callback(
-        null,
-        res.map(data => {
-          return {
-            ip: data[0]['networkInterfaces'][0]['accessConfigs']
-              ? data[0]['networkInterfaces'][0]['accessConfigs'][0]['natIP']
-              : 'no external ip',
-            name: data[0].name,
-          };
-        })
-      )
-    )
-    .catch(err => callback(err));
+async function listVms() {
+  try {
+    const [vms] = await zone.getVMs();
+    const results = [];
+    for (const i in vms) {
```

Comment:
This could be simplified a fair bit.  

```js
const [vms] = await zone.getVMs();
await Promise.all(
  vms.map(vm => {
    const [metadata] = await vm.getMetadata();
    return {
        ip: data[0]['networkInterfaces'][0]['accessConfigs']
          ? data[0]['networkInterfaces'][0]['accessConfigs'][0]['natIP']
          : 'no external ip',
        name: data[0].name,
      }
  })
)
```

Or something like that :)

Reply:
Done